src/arc.rs

```diff
@@ -119,6 +120,35 @@ impl<T: Eq + Hash + Send + Sync + 'static> ArcIntern<T> {
         };
         (*boxed).downcast_ref().unwrap()
     }
+    /// Intern a value from a reference with atomic reference counting.
+    ///
+    /// If this value has not previously been
+    /// interned, then `new` will allocate a spot for the value on the
+    /// heap and generate that value using `T::from(val)`.
+    pub fn from_ref<'a, Q: ?Sized + Eq + Hash + 'a>(val: &'a Q) -> ArcIntern<T>
+    where
+        T: Borrow<Q> + From<&'a Q>,
+    {
+        // No reference only fast-path as
+        // the trait `std::borrow::Borrow<Q>` is not implemented for `Arc<T>`
+        Self::new(val.into())
+    }
+    /// See how many objects have been interned.  This may be helpful
+    /// in analyzing memory use.
+    pub fn num_objects_interned() -> usize {
+        Self::get_container().len()
+    }
+    /// Return the number of counts for this pointer.
+    pub fn refcount(&self) -> usize {
+        unsafe { self.pointer.as_ref().count.load(Ordering::Acquire) }
```

Comment:
Unless you're doing something weird with this, we should be able to load relaxed.
```suggestion
        unsafe { self.pointer.as_ref().count.load(Ordering::Relaxed) }
```
I know you're just copying the old code here, but I try to notice and fix what I see when i see it.  Acquire would only be needed if we were to use this count as a guard somewhere, which I'm not seeing.

Reply:
I'm not super familiar with memory ordering, the way I see it rust std uses acquire ordering for `strong_count`, so I'm not super confident that relaxed ordering is fine here TBH. I'll leave this as is